cpp/include/cudf/io/functions.hpp

```diff
@@ -616,28 +766,140 @@ struct write_parquet_args {
   /// Set of columns to output
   table_view table;
   /// Optional associated metadata
-  const table_metadata* metadata;
+  const table_metadata* metadata = nullptr;
   /// Optionally return the raw parquet file metadata output
   bool return_filemetadata = false;
   /// Column chunks file path to be set in the raw output metadata
   std::string metadata_out_file_path;
 
+  explicit write_parquet_args(sink_info const& sink_, table_view const& table_)
+    : sink(sink_), table(table_)
```

Comment:
good opportunity to ditch the underscores.
```suggestion
  explicit write_parquet_args(sink_info const& sink, table_view const& table)
    : sink(sink), table(table)
```

Reply:
table will have issue with cudf::table